homeassistant/components/generic/camera.py

```diff
@@ -125,32 +125,44 @@ def camera_image(self):
         ).result()
 
     async def async_camera_image(self):
+        """Wrap _async_camera_image with an asyncio.shield."""
+        # Shield the request because of https://github.com/encode/httpx/issues/1461
+        try:
+            self._last_url, self._last_image = await asyncio.shield(
+                self._async_camera_image()
+            )
+        except asyncio.CancelledError:
+            _LOGGER.warning("Timeout getting camera image from %s", self._name)
```

Comment:
This is wrong though. Now you are swallowing the cancellation here. Any code further out doesn't get cancelled. You must reraise the exception.

Reply:
Thanks, made the change